homeworks/week2/Tam-Tran.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+# Time complexity: O(n):
+# - we iterate through the input string/array once
+# - look up to a checked hash once
+# - and append to the result array (the size of array is capped at string encoding size), we can improve this to use a link list
+#   which has a O(1) for add an item to it
+# Space complexity: O(1): the data structure we create is:
+# - a result string (it is capped at string encoding standard size)
+# - a hashmap collection of visited characters (capped at string encodinng size as well)
+def removeDuplicateCharacters(inputVal):
```

Comment:
nit: these camelCase code style is Java.

For python, I believe a more common naming would be
```
def remove_duplicate_characters(input_val):
```

Reply:
Yeah thanks for your useful feedback, havent code python nearly a year, should I correct it before PR can be merged ?